src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,417 @@
+}
+
+CHIP_ERROR OperationalStateServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (aPath.mAttributeId)
+    {
+    case OperationalState::Attributes::OperationalStateList::Id: {
+        OperationalStateStructDynamicList * pOpList = nullptr;
+        size_t size                                 = 0;
+        GetOperationalStateList(&pOpList, size);
+        if (size == 0)
+        {
+            err = aEncoder.EncodeNull();
+        }
+        else
+        {
+            return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+                for (OperationalStateStructDynamicList * pHead = pOpList; pHead != nullptr; pHead = pHead->Next)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(*pHead));
+                }
+                ReleaseOperationalStateList(pOpList);
+                pOpList = nullptr;
+                return CHIP_NO_ERROR;
+            });
+        }
+    }
+    break;
+
+    case OperationalState::Attributes::OperationalState::Id: {
+        OperationalState::Structs::OperationalStateStruct::Type opState;
+
+        opState.operationalStateID    = static_cast<OperationalState::OperationalStateEnum>(operationalState.OperationalStateID);
+        opState.operationalStateLabel = CharSpan::fromCharString(operationalState.OperationalStateLabel);
+
+        return aEncoder.Encode(opState);
+    }
+    break;
+
+    case OperationalState::Attributes::OperationalError::Id: {
+
+        OperationalState::Structs::ErrorStateStruct::Type opError;
+
+        opError.errorStateID = static_cast<OperationalState::ErrorStateEnum>(operationalError.ErrorStateID);
+        opError.errorStateLabel.SetNonNull(operationalError.ErrorStateLabel, strlen(operationalError.ErrorStateLabel));
+        opError.errorStateDetails.Emplace(operationalError.ErrorStateDetails, strlen(operationalError.ErrorStateDetails));
+
+        return aEncoder.Encode(opError);
+    }
+    break;
+
+    case OperationalState::Attributes::PhaseList::Id: {
+        PhaseListCharSpan * phaseList = nullptr;
+        size_t size                   = 0;
+        GetPhaseList(&phaseList, size);
+        if (size == 0)
+        {
+            err = aEncoder.EncodeNull();
+        }
+        else
+        {
+            return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+                for (PhaseListCharSpan * pHead = phaseList; pHead != nullptr; pHead = pHead->Next)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(pHead->phase));
+                }
+                ReleasePhaseList(phaseList);
+                phaseList = nullptr;
+                return CHIP_NO_ERROR;
+            });
+        }
+    }
+    break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OperationalStateServer::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    return CHIP_NO_ERROR;
```

Comment:
Why is this implemented at all?  It doesn't need to be implemented.

Reply:
Done. Remove it.